### PR TITLE
Handling of onSpinWait and its absence in AffinityPool

### PR DESCRIPTION
**Pull request by viktorklang**

### CODE REVIEW

File: akka-actor/src/main/scala/akka/dispatch/affinity/AffinityPool.scala [hunk 1]
```diff
@@ -85,10 +86,7 @@ private[affinity] object AffinityPool {
           idling = true
           transitionTo(Spinning)
         case Spinning =>
-          onSpinWaitMethodHandle match {
-            case OptionVal.Some(m) => m.invokeExact()
-            case OptionVal.None    =>
-          }
+          onSpinWaitMethodHandle.invokeExact(): Unit
```
Comment (viktorklang) [inline akka-actor/src/main/scala/akka/dispatch/affinity/AffinityPool.scala:89]: The type ascription here is crucial to get the `invokeExact` to work

File: akka-actor/src/main/scala/akka/dispatch/affinity/AffinityPool.scala [hunk 2]
```diff
@@ -43,10 +43,11 @@ private[affinity] object AffinityPool {
   final val Terminated = 5
 
   // Method handle to JDK9+ onSpinWait method
-  private val onSpinWaitMethodHandle =
-    try OptionVal.Some(MethodHandles.lookup.findStatic(classOf[Thread], "onSpinWait", methodType(classOf[Void])))
+  final val noopMethodHandle =  MethodHandles.constant(classOf[AnyRef], null).asType(methodType(Void.TYPE))
```
Comment (viktorklang) [inline akka-actor/src/main/scala/akka/dispatch/affinity/AffinityPool.scala:46]: This is the cleanest path to getting a no-op from my brief but intense search

**Comment by patriknw** [comment]
Is this a duplicate of https://github.com/akka/akka/pull/28946 ?

**Comment by viktorklang** [comment]
@patriknw One is for 2.5.x and one is for master

**Comment by patriknw** [comment]
ah, let's complete the one for master first and then backport

**Comment by viktorklang** [comment]
@patriknw Agreed—I just wanted to make sure that it would work on both sides. I'll update here once I find something which makes the `native-image` happy

**Comment by viktorklang** [comment]
@chbatey @patriknw Ready to merge on my part.

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
@viktorklang Can you update the PR for master with the same thing, or do you want me to take care of it? Can then be included in tomorrow's 2.6.5.

**Comment by viktorklang** [comment]
@patriknw I've pushed the same change to the 2.6 branch as well now

**Review by ennru — APPROVED** [review]
LGTM.

**Pull request merged**